pkg/resource/resource_test.go

```diff
@@ -273,7 +273,75 @@ spec:
 	})
 })
 
-var _ = Describe("creator_CheckForImagePullBackOff", func() {
+var _ = Describe("resource_GetObjectsFromYAML", func() {
+	var (
+		ctrl          *gomock.Controller
+		kubeClient    *clients.MockClientsInterface
+		mockLifecycle *lifecycle.MockLifecycle
+		metricsClient *metrics.MockMetrics
+		pollActions   *poll.MockPollActions
+		kernelData    *kernel.MockKernelData
+		proxyAPI      *proxy.MockProxyAPI
+		helper        *resourcehelper.MockHelper
+		scheme        *runtime.Scheme
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		kubeClient = clients.NewMockClientsInterface(ctrl)
+		mockLifecycle = lifecycle.NewMockLifecycle(ctrl)
+		metricsClient = metrics.NewMockMetrics(ctrl)
+		pollActions = poll.NewMockPollActions(ctrl)
+		kernelData = kernel.NewMockKernelData(ctrl)
+		proxyAPI = proxy.NewMockProxyAPI(ctrl)
+		helper = resourcehelper.NewMockHelper(ctrl)
+		scheme = runtime.NewScheme()
+		err := v1.AddToScheme(scheme)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	yamlSpec := []byte(`---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: podName
+spec:
+  containers:
+  - name: podName
+    image: podimage:1.14.2
+    ports:
+    - containerPort: 80
+  restartPolicy: Always
+---
+apiVersion: v1
+kind: DaemonSet
+metadata:
+  name: dsName
+spec:
+  containers:
+  - name: dsName
+    image: dsimage:1.14.2
+    ports:
+    - containerPort: 80
+  restartPolicy: Always
+`)
+	It("good flow", func() {
+		r := NewResourceAPI(kubeClient, metricsClient, pollActions, kernelData, scheme, mockLifecycle, proxyAPI, helper)
+		resultList, err := r.GetObjectsFromYAML(context.TODO(), yamlSpec)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(resultList.Items)).To(Equal(2))
+		for _, item := range resultList.Items {
+			Expect(item.GetName()).To(BeElementOf([]string{"dsName", "podName"}))
+			Expect(item.GetKind()).To(BeElementOf([]string{"DaemonSet", "Pod"}))
+		}
```

Comment:
I think the order should be deterministic here?
```suggestion
		Expect(resultList.Items[0].GetName()).To(Equal("podName"))
		Expect(resultList.Items[0].GetKind()).To(Equal("Pod"))
		Expect(resultList.Items[1].GetName()).To(Equal("dsName"))
		Expect(resultList.Items[1].GetKind()).To(Equal("DaemonSet"))
```

Reply:
fixed